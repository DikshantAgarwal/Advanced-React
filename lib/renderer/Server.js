import React from 'react';
import ReactDomServer from 'react-dom/server';
import axios from 'axios';
import config from '../config';
import DataApi from '../DataApi'

import App from 'component/App';

// console.log(config)
const ServerRender = async () => {

    const res = await axios.get(`http://${config.host}:${config.port}/data`)
    const api = new DataApi(res.data)
    console.log(api)

    const initialData = {
        articles: api.getArticles(),
        authors: api.getAuthores()
    }

    return {
        initialMarkup: ReactDomServer.renderToString(
            <App initialData={initialData} />
        ),
        initialData,
        }
}

export default ServerRender;