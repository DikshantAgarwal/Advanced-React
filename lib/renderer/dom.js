import React, { Component } from 'react'
import ReactDom from 'react-dom';
import App from 'component/App';

// const initialData={
//     articles:{},
//     authors:{},
// }
 

ReactDom.render(
    <App initialData={window.initialData}/>,
    document.getElementById('root')
)