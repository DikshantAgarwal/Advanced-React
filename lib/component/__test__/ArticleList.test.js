import React from 'react';
import ArticleList from '../ArticleList';
import renderer from 'react-test-renderer';

describe('ArticleList ',()=>{

    const testProps={
        articles:{
            a:{id:'a'},
            b:{id:'b'},
        },
        store:{
            lookUpAuthor: jest.fn(()=>({})),   
        }
    }
    it('Should render properly',()=>{
        const tree= renderer.create(
            <ArticleList
                {...testProps}
            />
        ).toJSON();
        // console.log(tree);
        expect(tree).toMatchSnapshot();
        expect(tree.children.length).toBe(2);
    })
})