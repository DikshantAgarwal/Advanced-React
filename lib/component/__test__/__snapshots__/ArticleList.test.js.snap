// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList  Should render properly 1`] = `
<div>
  <div
    style={
      Object {
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={
        Object {
          "paddingBottom": "10",
          "paddingTop": "10",
        }
      }
    >
      <a />
    </div>
    <div
      style={
        Object {
          "paddingLeft": "20",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={
        Object {
          "paddingBottom": "10",
          "paddingTop": "10",
        }
      }
    >
      <a />
    </div>
    <div
      style={
        Object {
          "paddingLeft": "20",
        }
      }
    />
  </div>
</div>
`;
